src/chains/arbitrum/signatureTypes.ts

```diff
@@ -0,0 +1,110 @@
+  ],
+};
+
+const arbitrumUnsignedTx: SignatureType = {
+  prefixByte: 0x65,
+  description:
+    "A message from a user on L1 to a contract on L2 that uses the bridge for authentication instead of requiring the user's signature.",
+  signedData: [
+    'keccak256(0x65 || rlp([chainId, from, nonce, maxFeePerGas, gasLimit, to, value, data]))',
+  ],
+  signs: 'transaction',
+  references: [
+    txTypeDocs,
+    'https://github.com/OffchainLabs/go-ethereum/blob/master/core/types/transaction.go#L49',
+    'https://github.com/OffchainLabs/go-ethereum/blob/master/core/types/arb_types.go#L43-L53',
+  ],
+};
+
+const arbitrumContractTx: SignatureType = {
+  prefixByte: 0x66,
+  description:
+    "Similar to an `ArbitrumUnsignedTx` but intended for smart contracts. Uses the bridge's unique, sequential nonce rather than requiring the caller specify their own.",
+  signedData: [
+    'keccak256(0x66 || rlp([chainId, requestId, from, maxFeePerGas, gasLimit, to, value, data]))',
+  ],
+  signs: 'transaction',
+  references: [
+    txTypeDocs,
+    'https://github.com/OffchainLabs/go-ethereum/blob/master/core/types/transaction.go#L50',
+    'https://github.com/OffchainLabs/go-ethereum/blob/master/core/types/arb_types.go#L104-L114',
+  ],
+};
+
+const arbitrumSubmitRetryableTx: SignatureType = {
+  prefixByte: 0x69,
+  description: 'A retryable submission and may schedule an ArbitrumRetryTx if provided enough gas.',
+  signedData: [
+    'keccak256(0x69 || rlp([chainId, requestId, from, l1BaseFee, depositValue, maxFeePerGas, gasLimit, retryTo, retryValue, beneficiary, maxSubmissionFee, feeRefundAddress, retryData]))',
+  ],
+  signs: 'transaction',
+  references: [
+    txTypeDocs,
+    'https://github.com/OffchainLabs/go-ethereum/blob/master/core/types/transaction.go#L52',
+    'https://github.com/OffchainLabs/go-ethereum/blob/master/core/types/arb_types.go#L232-L247',
+  ],
+};
+
+const arbitrumRetryTx: SignatureType = {
+  prefixByte: 0x68,
+  description:
+    'Transactions scheduled by calls to the redeem precompile method and via retryable auto-redemption.',
+  signedData: [
+    'keccak256(0x68 || rlp([chainId, nonce, from, maxFeePerGas, gasLimit, to, value, data, ticketId, refundTo, maxRefund, submissionFeeRefund]))',
+  ],
+  signs: 'transaction',
+  references: [
+    txTypeDocs,
+    'https://github.com/OffchainLabs/go-ethereum/blob/master/core/types/transaction.go#L51',
+    'https://github.com/OffchainLabs/go-ethereum/blob/master/core/types/arb_types.go#L161-L176',
+  ],
+};
+
+const arbitrumInternalTx: SignatureType = {
+  prefixByte: 0x6a,
+  description: 'ArbOS-created transaction to update state between user-generated transactions.',
+  signedData: ['keccak256(0x6a || rlp([chainId, data]))'],
+  signs: 'transaction',
+  references: [
+    txTypeDocs,
+    'https://github.com/OffchainLabs/go-ethereum/blob/master/core/types/transaction.go#L53',
+    'https://github.com/OffchainLabs/go-ethereum/blob/master/core/types/arb_types.go#L387-L390',
```

Comment:
These line numbers are already broken, the file changed yesterday. I'd recommend pinning links to a commit or release tag.

Reply:
Heh was wondering if these files were stable enough to link to master directly to avoid clicking a permalink and having to wonder if that link is actually up to date. Seems the answer is no, so back to permalinks 😅